### DISCUSSION TITLE
Capacitor support

### DISCUSSION THREAD

**Discussion by chris-si** [electric-sql/electric#1434] [Feature requests]
Since I couldn't reopen the old discussion #1424 of the now 'legacy' system, here it is again :-)

Currently there doesn't seem to be an official driver/plugin to use electric-sql with capacitor. The mobile docs only mention Expo and React Native atm. It would be great if capacitor apps would also be officially supported with 'native' drivers like in [sqlite](https://github.com/capacitor-community/sqlite) so data could be synced to pglite(?) on a mobile device and stored via the native (not web) layer.

**Comment by KyleAMathews** [comment]
👋 we haven't tested it yet but the new `Shape` API should be pretty easy to integrate with any environment that supports JS — https://next.electric-sql.com/api/clients/typescript#shape

Would you be interested in creating a little example?

**Comment by jack-hanlon** [comment]
Yes !! I'm looking into this as well, does a Capacitor plugin need to be made for the PGLite DB that integrates with Electric SQL. This would be a fantastic alternative to using SQLite, seeing as my cloud db is postgres and I wouldn't have to change my schemas
